ddl/ddl_api.go

```diff
@@ -3542,3 +3576,23 @@ type lockTablesArg struct {
 	SessionInfo   model.SessionInfo
 	IsCleanup     bool
 }
+
+// When handle charset and collate of a column, we should take collate in option into consideration
+// rather than handling it separately. Then the following option processing logic should ignores this
+// cause we will delete it from option here
+func extractCollateFromOption(def *ast.ColumnDef) string {
+	specifiedCollate := ""
+	collateIndex := -1
+	for i, op := range def.Options {
```

Comment:
As you may see, it is possible that we encounter multiple `ast.ColumnOptionCollate`, we should choose the last one as the final result:

For example, in MySQL 8.0:
```
mysql> create table tm(a varchar(20) collate utf8_unicode_ci collate utf8_general_ci) default charset=utf8mb4;
Query OK, 0 rows affected, 2 warnings (0.08 sec)

mysql> show create table tm;
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                     |
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tm    | CREATE TABLE `tm` (
  `a` varchar(20) CHARACTER SET utf8 COLLATE utf8_general_ci DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci |
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)
```
Notice that collation of column `a` is `utf8_general_ci`.

Please also notice that the when we resolve multiple collations, their charsets must be the same:

```
mysql> create table tm(a varchar(20) collate ascii_bin collate utf8_general_ci) default charset=utf8mb4;
ERROR 1253 (42000): COLLATION 'utf8_general_ci' is not valid for CHARACTER SET 'ascii'
```

Reply:
You can follow similar existing logics to handle multiple `ColumnOptionCollate`:

https://github.com/pingcap/tidb/blob/b3f71868c159ba33a8b88956a828d06d43a5eb6e/ddl/ddl_api.go#L109

Reply:
> You can follow similar existing logics to handle multiple `ColumnOptionCollate`:
> 
> https://github.com/pingcap/tidb/blob/b3f71868c159ba33a8b88956a828d06d43a5eb6e/ddl/ddl_api.go#L109

That's great !!！
From mysql syntax, charset is unique specified, but collate can be multiple. 
That's means I need to follow mysql's behavior when handling multiple collate.

And this reference code do me a great favor!!!